rosetta-wallet/src/main.rs

```diff
@@ -79,6 +92,33 @@ async fn main() -> Result<()> {
             let balance = wallet.balance().await?;
             println!("{}", rosetta_client::amount_to_string(&balance)?);
         }
+        Command::DeployContract(DeployContractOpts { bin_path }) => {
+            match wallet.config().blockchain {
+                "bitcoin" => {
+                    anyhow::bail!("Not implemented");
+                }
+                _ => {
+                    let compiled_contract_str = read_to_string(bin_path.to_string())?;
+                    let compiled_contract_bin = compiled_contract_str
+                        .strip_suffix('\n')
+                        .unwrap_or(&compiled_contract_str);
+                    let bytes = hex::decode(compiled_contract_bin).unwrap();
+                    let response = wallet.eth_deploy_contract(bytes.to_vec()).await?;
+                    println!("Deploy contract response {:?}", response);
```

Comment:
this will be annoying to parse in a script. you should just print the smart contract address